src/script/microbit-interfacing/state-updaters.ts

```diff
@@ -0,0 +1,188 @@
+/**
+ * (c) 2023, Center for Computational Thinking and Design at Aarhus University and contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+import { get } from 'svelte/store';
+import { DeviceRequestStates } from '../stores/connectDialogStore';
+import { ModelView, state } from '../stores/uiStore';
+import { Paths, currentPath, navigate } from '../../router/paths';
+import MBSpecs from './MBSpecs';
+import StaticConfiguration from '../../StaticConfiguration';
+import { sendToOutput } from './microbit-bluetooth';
+import { HexOrigin } from './Microbits';
+
+// TODO: We've lost reconnect and logging timeouts
+// TODO: We've lost logging in general, but most of it was unhelpful.
```

Comment:
Flagging todos.

Reply:
The reconnect timeout seems a bit meaningless. It doesn't appear to be the amount of time we try to reconnect for after a connection loss.